src/com/sap/piper/CommonPipelineEnvironment.groovy

```diff
@@ -0,0 +1,154 @@
+        return valueMap.get(property)
+    }
+
+    def setAppContainerProperty(property, value) {
+        appContainerProperties[property] = value
+    }
+
+    def getAppContainerProperty(property) {
+        return appContainerProperties[property]
+    }
+
+    // goes into measurement jenkins_custom_data
+    def setInfluxCustomDataEntry(key, value) {
+        InfluxData.addField('jenkins_custom_data', key, value)
+    }
+    // goes into measurement jenkins_custom_data
+    @Deprecated // not used in library
+    def getInfluxCustomData() {
+        return InfluxData.getInstance().getFields().jenkins_custom_data
+    }
+
+    // goes into measurement jenkins_custom_data
+    def setInfluxCustomDataTagsEntry(key, value) {
+        InfluxData.addTag('jenkins_custom_data', key, value)
+    }
+    // goes into measurement jenkins_custom_data
+    @Deprecated // not used in library
+    def getInfluxCustomDataTags() {
+        return InfluxData.getInstance().getTags().jenkins_custom_data
+    }
+
+    void setInfluxCustomDataMapEntry(measurement, field, value) {
+        InfluxData.addField(measurement, field, value)
+    }
+    @Deprecated // not used in library
+    def getInfluxCustomDataMap() {
+        return InfluxData.getInstance().getFields()
+    }
+
+    def setInfluxCustomDataMapTagsEntry(measurement, tag, value) {
+        InfluxData.addTag(measurement, tag, value)
+    }
+    @Deprecated // not used in library
+    def getInfluxCustomDataMapTags() {
+        return InfluxData.getInstance().getTags()
+    }
+
+    @Deprecated // not used in library
+    def setInfluxStepData(key, value) {
+        InfluxData.addField('step_data', key, value)
+    }
+    @Deprecated // not used in library
+    def getInfluxStepData(key) {
+        return InfluxData.getInstance().getFields()['step_data'][key]
+    }
+
+    @Deprecated // not used in library
+    def setInfluxPipelineData(key, value) {
+        InfluxData.addField('pipeline_data', key, value)
+    }
+    @Deprecated // not used in library
+    def setPipelineMeasurement(key, value){
+        setInfluxPipelineData(key, value)
+    }
+    @Deprecated // not used in library
+    def getPipelineMeasurement(key) {
+        return InfluxData.getInstance().getFields()['pipeline_data'][key]
+    }
+
+    def reset() {
+        appContainerProperties = [:]
+        configuration = [:]
+        artifactVersion = null
+
+        gitCommitId = null
+        gitCommitMessage = null
+        gitSshUrl = null
+        gitHttpsUrl = null
+        gitBranch = null
+
+        githubOrg = null
+        githubRepo = null
+
+        mtarFilePath = null
+        valueMap = [:]
+
+        changeDocumentId = null
+
+        InfluxData.reset()
+    }
+
+    Map getStepConfiguration(stepName, stageName = env.STAGE_NAME, includeDefaults = true) {
+        Map defaults = [:]
+        if (includeDefaults) {
+            defaults = DefaultValueCache.getInstance()?.getDefaultValues()?.general ?: [:]
```

Comment:
you changed it from 
`defaults = ConfigurationLoader.defaultGeneralConfiguration()`
why?

Reply:
With that the ConfigurationLoader is not in the game here.